pkg/jx/cmd/get_applications.go

```diff
@@ -243,6 +261,31 @@ func (o *GetApplicationsOptions) generateTable(apps []string, envApps []EnvApps,
 	return table
 }
 
+// BuildGitURL concatenates various fields of a SourceRepository CRD into a valid URL.
+func BuildGitURL(sourceRepository *v1.SourceRepository) string {
+	if sourceRepository.Spec.Provider == "" || sourceRepository.Spec.Org == "" || sourceRepository.Spec.Repo == "" {
+		return "None Found"
+	}
+	return sourceRepository.Spec.Provider + "/" + sourceRepository.Spec.Org + "/" + sourceRepository.Spec.Repo + ".git"
+}
+
+// GetSourceRepositoryForApplication fetches a SourceRepository CRD for a particular application by trying to match its name to a repo.
+func GetSourceRepositoryForApplication(application string, sourceRepositories altv1.SourceRepositoryInterface) (*v1.SourceRepository, error) {
+	// We can't pull the SourceRepository by name because we don't know the org, so we iterate over all of them and match the application name to the repo.
+	repoList, err := sourceRepositories.List(metav1.ListOptions{})
+	if err != nil {
+		return nil, err
```

Comment:
I would wrap this error with some context.

Reply:
Will do,